rebasehelper/cli.py

```diff
@@ -148,15 +159,103 @@ def __getattr__(self, name):
+            shutil.rmtree(results_dir)
+        os.makedirs(results_dir)
+        os.makedirs(os.path.join(results_dir, settings.REBASE_HELPER_LOGS))
+        os.makedirs(os.path.join(results_dir, 'old'))
+        os.makedirs(os.path.join(results_dir, 'new'))
+
+    @staticmethod
+    def _add_debug_log_file(results_dir):
+        debug_log_file = os.path.join(results_dir, settings.REBASE_HELPER_DEBUG_LOG)
+        try:
+            LoggerHelper.add_file_handler(logger,
+                                          debug_log_file,
+                                          logging.Formatter("%(asctime)s %(levelname)s\t%(filename)s"
+                                                            ":%(lineno)s %(funcName)s: %(message)s"),
+                                          logging.DEBUG)
+        except (IOError, OSError):
+            logger.warning("Can not create debug log '%s'", debug_log_file)
+        else:
+            return debug_log_file
+
+    @staticmethod
+    def _add_report_log_file(results_dir):
+        report_log_file = os.path.join(results_dir, settings.REBASE_HELPER_REPORT_LOG)
+        try:
+            LoggerHelper.add_file_handler(logger_report,
+                                          report_log_file,
+                                          None,
+                                          logging.INFO)
+        except (IOError, OSError):
+            logger.warning("Can not create report log '%s'", report_log_file)
+        else:
+            return report_log_file
+
+    @staticmethod
+    def setup(arguments):
+        # be verbose until debug_log_file is created
+        handler = LoggerHelper.add_stream_handler(logger, logging.DEBUG)
+
+        cli = CLI(arguments)
+
+        execution_dir = cli.results_dir if cli.results_dir else os.getcwd()
+        results_dir = os.path.join(execution_dir, settings.REBASE_HELPER_RESULTS_DIR)
+
+        # if not continuing, check the results dir
+        if not cli.cont and not cli.build_only and not cli.comparepkgs:
+            CliHelper._check_results_dir(results_dir)
+
+        # This is used if user executes rebase-helper with --continue
+        # parameter even when directory does not exist
+        if not os.path.exists(results_dir):
+            os.makedirs(results_dir)
+            os.makedirs(os.path.join(results_dir, settings.REBASE_HELPER_LOGS))
+
+        debug_log_file = CliHelper._add_debug_log_file(results_dir)
+        report_log_file = CliHelper._add_report_log_file(results_dir)
+
+        if not cli.verbose:
+            handler.setLevel(logging.INFO)
+
+        return cli, execution_dir, debug_log_file, report_log_file
+
+    @staticmethod
     def run():
         try:
-            cli = CLI(sys.argv[1:])
-            app = Application(cli)
+            debug_log_file = None
+            cli, execution_dir, debug_log_file, report_log_file = CliHelper.setup(sys.argv[1:])
```

Comment:
We don't need to pass the ```sys.argv``` to the parser. The possibility to pass the arguments explicitly is more interesting for testing.